#!/usr/bin/env python
PACKAGE = "sr_grasp_mesh_planner"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("max_grasps", int_t, 0,
        "The number of desired grasps.",
	2, 1, 1000)

gen.add("timeout_one_grasp", double_t, 0,
        "The maximum time (in seconds) allowed for the grasp planning of one grasp. "
	"No limit if zero is given.",
	60.0, 0.0, 3600.0)

gen.add("min_quality", double_t, 0,
        "The minimum quality for a grasp.",
	0.2, 0.0, 10.0)

gen.add("force_closure", bool_t, 0,
        "Force closure grasps or not?",
	True)

approach_movement_enum = gen.enum([ gen.const("bounding_box", int_t, 0, "Bounding box based approach movement generator"),
                                    gen.const("surface_normal", int_t, 1, "Object surface normal based approach movement generator") ],
                                  "An enum to set approach movement generator")

gen.add("approach_movement", int_t, 0,
        "An approach movement generator parameter which is edited via an enum",
        0, 0, 1, edit_method=approach_movement_enum)

exit(gen.generate(PACKAGE, "sr_grasp_mesh_planner", "Planner"))
